File: templates/templates/CliModuleNode/module.js
#!/usr/bin/env node
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

const directory = dirname(fileURLToPath(import.meta.url));
if (process.argv.slice(2).length === 1 && process.argv[2] === "--corevar-describe") {
  console.log(readFileSync(join(directory, "corevar.module.json"), "utf8"));
  process.exit(0);
}
const args = process.argv.slice(2);
console.log(`Hello, ${args[0] === "hello" && args[1] ? args[1] : "world"}!`);

File: templates/templates/CliModulePython/module.py
#!/usr/bin/env python3
import json
import pathlib
import sys

if sys.argv[1:] == ["--corevar-describe"]:
    print(pathlib.Path(__file__).with_name("corevar.module.json").read_text(encoding="utf-8"))
    raise SystemExit(0)

name = sys.argv[2] if len(sys.argv) > 2 and sys.argv[1] == "hello" else "world"
print(f"Hello, {name}!")
